www/%username/settings/edit.spt

```diff
@@ -25,8 +28,47 @@ if 'new-password' in body:
         )
         if not p:
             raise response.error(403, _("Incorrect password"))
-    p.update_password(body['new-password'])
-    out['msg'] = _("Your password has been changed.")
+
+    if body.get('confirmed') != 'true':
+        passhash = sha1(body['new-password'].encode("utf-8")).hexdigest().upper()
+        passhash_short = passhash[:5]
+        URL = "https://api.pwnedpasswords.com/range/"+passhash_short
+
+        r = requests.get(url = URL)
+        count = 0
+        for line in r.text.split("\n"):
+            suffix = line.split(":")[0]
+            if passhash_short + suffix == passhash:
+                count = int(line.split(":")[1].strip())
+        strength, improvements = passwordmeter.test(body['new-password'])
```

Comment:
This is code duplication, it should be replaced with a call to a separate `check_password` function.